module/VuFindSearch/src/VuFindSearch/Backend/Blender/Backend.php

```
@@ -0,0 +1,590 @@
+
+            $offsetOk = $backendOffsets[$backendAtPos]
+                < $backendTotals[$backendAtPos];
+            $record = $offsetOk ? $this->getRecord(
+                $backendDetails[$backendAtPos],
+                $backendRecords[$backendAtPos],
+                $backendOffsets[$backendAtPos]++,
+                $blockSize
+            ) : null;
+
+            if (null === $record) {
+                // Try other backends:
+                foreach ($availableBackendIds as $backendId) {
+                    if ($backendId === $backendAtPos) {
+                        continue;
+                    }
+                    $offsetOk = $backendOffsets[$backendId]
+                        < $backendTotals[$backendId];
+                    $record = $offsetOk ? $this->getRecord(
+                        $backendDetails[$backendId],
+                        $backendRecords[$backendId],
+                        $backendOffsets[$backendId]++,
+                        $blockSize
+                    ) : null;
+
+                    if (null !== $record) {
+                        break;
+                    }
+                }
+            }
+
+            if (null === $record) {
+                break;
+            }
+            $mergedCollection->add($record, false);
+        }
+    }
+
+    /**
+     * Retrieve a single document.
+     *
+     * @param string   $id     Document identifier
+     * @param ParamBag $params Search backend parameters
+     *
+     * @return \VuFindSearch\Response\RecordCollectionInterface
+     */
+    public function retrieve($id, ParamBag $params = null)
+    {
+        throw new \Exception('Blender does not support retrieve');
+    }
+
+    /**
+     * Return the record collection factory.
+     *
+     * Lazy loads a generic collection factory.
+     *
+     * @return RecordCollectionFactoryInterface
+     */
+    public function getRecordCollectionFactory()
+    {
+        return null;
```

Any particular reason to return null here? That doesn't seem technically legal according to the `@return` type. Should this throw an exception instead if the feature is unsupported?

---

Changed. I'm not even sure why this is a public function in the abstract class.

---

It's used in tests, so that may be the primary reason.